### ISSUE TITLE
pretty-printing of extension-constructors appears broken

### ISSUE THREAD

**Issue by chetmurthy** [ocaml/ocaml#9588]
Ocaml version: 4.10.0

The pretty-printing of rebound extension-constructors appears broken.  Here's an example:
```
# {| type t += A = M.A [@a] |} |> Official.Implem.pa |> Official.Implem.pr |> print_string ;;
type t +=  
  | A[@a ] = M.A- : unit = ()
# {| type t += A [@a] = M.A |} |> Official.Implem.pa |> Official.Implem.pr |> print_string ;;
Exception: Syntaxerr.Error (Syntaxerr.Other <loc>).
Raised at file "parsing/parse.ml", line 65, characters 11-49
Called from unknown location
Called from file "toplevel/toploop.ml", line 212, characters 17-27
# 
```
[`Official.Implem.pa` and `Official.Implem.pr` are wrappers around functions in compiler-libs.]

A quick perusal of `parser.mly` (line 3047) and `pprintast.ml` (line 1573) appears to confirm.  I'm going to presume that the parser is correct and the pretty-printer is wrong, and will patch to fix, but figure I should at least check that this assumption is correct.

**Comment by chetmurthy** [comment]
Sorry, my first attempt was based on 4.11; then I re-did it based on trunk.  Hope that was the right thing to do.

**Comment by damiendoligez** [comment]
Fixed by #9590
